manta-trusted-setup/src/groth16/ceremony/client.rs

```diff
@@ -130,10 +136,28 @@ where
         C::Identifier: Serialize,
         C::Nonce: DeserializeOwned,
     {
-        let (metadata, nonce) = client
+        let mut client_data = client
             .post("start", &identifier)
             .await
-            .map_err(into_ceremony_error)?;
+            .map_err(into_ceremony_error);
+        let term = Term::stdout();
+        let mut counter = 0u8;
+        println!("Connecting server for Metadata");
+        while let Err(CeremonyError::NotRegistered) = client_data {
+            if counter >= 60 {
+                panic!("This is taking longer than expected, please try again later.");
```

Comment:
I think it's reasonable to panic here but let's wait longer, like at least 10 minutes.